segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/operations/MergeSegmentOperation.java

```diff
@@ -156,12 +181,42 @@ private void write00(MergeSegmentOperation o, RevisionDataOutput target) throws
             target.writeLong(o.streamSegmentOffset);
         }
 
+        private void write01(MergeSegmentOperation o, RevisionDataOutput target) throws IOException {
+            if (o.attributeUpdates == null || o.attributeUpdates.isEmpty()) {
+                return;
+            }
+            target.length(target.getCollectionLength(o.attributeUpdates.size(), ATTRIBUTE_UUID_UPDATE_LENGTH));
+            target.writeCollection(o.attributeUpdates, this::writeAttributeUpdateUUID00);
+        }
+
+        private void writeAttributeUpdateUUID00(RevisionDataOutput target, AttributeUpdate au) throws IOException {
+            target.writeLong(au.getAttributeId().getBitGroup(0));
+            target.writeLong(au.getAttributeId().getBitGroup(1));
+            target.writeByte(au.getUpdateType().getTypeId());
```

Comment:
FYI To save space, you don't need this field. It is completely ignored upon recovery (as the validation happens once, before it's actually serialized). Since it's only 1 byte (and since appends already have it - I forgot why), then it's really up to you to decide whether to keep it or not.

Same with `getComparisonValue`. This would be a total of 9 bytes you can reduce this by.

Reply:
Ok, good to know, but as it is such a tiny piece of data, I think that the clarity in the code worth it to keep.